chapter_12_dependency_injection.asciidoc

```diff
@@ -631,7 +681,10 @@ rest of it.
 
 <3> And our messagebus is now a specific instance rather than the module global.
 
-// (ej) The "Flask-onic" version of bootstrap is the application factory pattern w/ blueprints.
+// TODO (EJ3) bus is still a module global, it's now just an instance of an class instead of a module.
```

Comment:
oh there's some confusing terminology here. the `bus` variable is still a global in the flask app module scope, but before, when we did `import messagebus`, we imported the module object itself, which is a process-level global/singleton.

Reply:
Yes, that's right.   A module level variable is effectively the same thing since it will behave like a [singleton](https://subscription.packtpub.com/book/application_development/9781783283378/2/ch02lvl1sec14/a-module-level-singleton).

Where this might matter is when the app using the bus is the unit test runner.  Importing `flask.app` will initialize it, when then sometimes leads people to patch it because they cannot control its initialization.

I dislike module singletons for this reason because they IHMO are another potential footgun, although it's a fairly common thing to do with wsgi apps. (Nit-picking done, no need to change the text.)